cli/pkg/kctrl/cmd/app/init/configure/fetch/fetch.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package fetch
+
+import (
+	"github.com/cppforlife/go-cli-ui/ui"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/app/init/build"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/app/init/common"
+	cmdcore "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/core"
+	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
+)
+
+type FetchStep struct {
+	ui                        cmdcore.IAuthoringUI
+	appBuild                  *build.AppBuild
+	isAppCommandRunExplicitly bool
+}
+
+func NewFetchStep(ui cmdcore.IAuthoringUI, appBuild *build.AppBuild, isAppCommandRunExplicitly bool) *FetchStep {
```

Comment:
Reason why we are having AppBuild in the Fetch step: Fetch step populates the AppBuild annotation(CurrentFetchOptionSelected). This annotation is being used later in the Template step as well to decide whether to add `helmTemplate` or not. We are passing AppBuild even in case of Template Step.

The alternative is to expose a method which returns the current fetch option. This method can be used by init.go to pass this information while instantiating `Template Step` in the init.go.